### PR TITLE
Sideways jumpthrus

### LINKED ISSUE #20: Multi-directional jumpthrus
> multi-directional jumpthru's, they'd be able to use any jumpthru tileset and just rotate it hopefully

_Requested by RadleyMcTuneston on collabcord._

### PR DESCRIPTION
**Pull request by max4805**
Closes #20.

This time, I didn't make it extend JumpThru, since those have little in common (... the _orientation_ to begin with: jump thrus extend on their width).

Most of how it works is an IL patch applied on multiple methods, that adds collide checks against sideways jumpthrus everywhere the game checks for solids. This way, Madeline can climb on and walljump off sideways jumpthrus.

(Note to self: the Julia part seems to need a tab to space pass.)

### CODE REVIEW

File: Entities/SidewaysJumpThru.cs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                if (jumpThru != null && jumpThru.AllowLeftToRight != movingLeftToRight) {
+                    self.Speed.X = 0;
+                }
+            }
+
+            return result;
+        }
+
+
+
```
Comment (0x0ade) [inline Entities/SidewaysJumpThru.cs]: What's up with the whitespace here?

Comment (max4805) [reply Entities/SidewaysJumpThru.cs]: That was to separate hook code from the entity code 😅 I think I should drop a comment in here instead.

**Comment by max4805** [comment]
I'll double-check what the base game does for orientation settings. Would "right" mean the upper side of the jumpthru is on the right side, so you can go left to right? 🤔

**Comment by max4805** [comment]
Attribute renamed on the Ahorn plugin side, comment added instead of that awkward gap on the C# side.

**Pull request merged**